### PR TITLE
Schematic description and stock

### PR DESCRIPTION
**Pull request by Kasmadei**
Resolves: https://github.com/kbss-cvut/fta-fmea-ui/issues/298
Resolves: https://github.com/kbss-cvut/fta-fmea-ui/issues/299

### CODE REVIEW

File: public/locales/cs/translation.json [hunk 1]
```diff
@@ -91,7 +92,8 @@
     "ataSystem": "ATA systém",
     "partNumber": "Číslo součásti",
     "stock": "Zásoba",
-    "quantity": "Množství"
+    "quantity": "Množství",
+    "schematicDescription": "Schematický popis"
```
Comment (blcham) [inline public/locales/cs/translation.json:96]: "schematicDescription": "Schematický popis"
->
"schematicDesignation": "Schématické označení"


Is there any point to do schematicDesignation -> schematicDescription ? To me the terms means little bit different thing.

File: public/locales/en/translation.json [hunk 2]
```diff
@@ -91,7 +92,8 @@
     "ataSystem": "ATA System",
     "partNumber": "Part number",
     "stock": "Stock",
-    "quantity": "Quantity"
+    "quantity": "Quantity",
+    "schematicDescription": "Schematic description"
```
Comment (blcham) [inline public/locales/en/translation.json:96]: see comment above

File: src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx [hunk 3]
```diff
@@ -33,6 +33,7 @@ const FaultEventMenu = ({ shapeToolData, onEventUpdated, refreshTree, rootIri }:
   const [partNumber, setPartNumber] = useState<string | undefined>(undefined);
   const [stock, setStock] = useState<string | undefined>(undefined);
   const [quantity, setQuantity] = useState<number | undefined>(undefined);
+  const [schematicDescription, setSchematicDescription] = useState<string | undefined>(undefined);
```
Comment (blcham) [inline src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx:36]: see comment above

File: src/models/eventModel.tsx [hunk 4]
```diff
@@ -51,6 +51,7 @@ const ctx = {
   editor: VocabularyUtils.PREFIX + "editor",
   username: VocabularyUtils.PREFIX + "username",
   estimate: VocabularyUtils.PREFIX + "has-estimate",
+  schematicDescription: VocabularyUtils.PREFIX + "schematic-designation",
```
Comment (blcham) [inline src/models/eventModel.tsx:54]: see comment above

**Review by blcham — CHANGES_REQUESTED** [review]
See my comment.

File: src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx [hunk 5]
```diff
@@ -68,27 +69,38 @@ const FaultEventMenu = ({ shapeToolData, onEventUpdated, refreshTree, rootIri }:
...
+    }
   }, [shapeToolData]);
 
   const basedFailureRate = shapeToolData?.supertypes?.supertypes?.hasFailureRate?.estimate?.value;
   const requiredFailureRate = shapeToolData?.supertypes?.hasFailureRate?.requirement?.upperBound;
 
   return (
     <Box paddingLeft={2} marginRight={2}>
-      <FaultEventShapeToolPane data={shapeToolData} onEventUpdated={onEventUpdated} refreshTree={refreshTree} />
+      {/* Hidden to prevent app crush after root node select (Bug with UI). Will be fixed in next PR */}
```
Comment (blcham) [inline src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx:102]: - put it to separate PR @Kasmadei

File: public/locales/cs/translation.json (hunk 1 above)
Comment (Kasmadei) [reply public/locales/cs/translation.json:96]: Changed.

File: public/locales/en/translation.json (hunk 2 above)
Comment (Kasmadei) [reply public/locales/en/translation.json:96]: Changed.

File: src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx (hunk 3 above)
Comment (Kasmadei) [reply src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx:36]: Changed.

File: src/models/eventModel.tsx (hunk 4 above)
Comment (Kasmadei) [reply src/models/eventModel.tsx:54]: Changed.

File: src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx (hunk 5 above)
Comment (Kasmadei) [reply src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx:102]: Removed from this PR

Comment (blcham) [reply src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx]: @Kasmadei i did not see any issue for that so i created it, i thought you will create PR with this commit as temporal workaround for the issue
